### ISSUE TITLE
`find_essential` returns ten identity matrices when a 5-point sample has no real solution, while complex slots are otherwise NaN

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4883]
`find_essential` (`run_5point`) returns `(B, 10, 3, 3)`: one slot per root of the degree-10 polynomial, unit-norm for a real root and `NaN` for a complex root. When *all ten* roots of a sample are complex, `null_to_Nister_solution` returns an all-NaN element and `run_5point` replaces it with ten `3 x 3` identity matrices. The identity is finite, has Frobenius norm `sqrt(3)` and is not an essential matrix (its singular values are `1, 1, 1`), so a caller that filters candidates with `isfinite` or `isnan` — the documented way to drop the complex slots — keeps ten fake solutions for that sample. Such samples exist: 2 of 4000 random 5-point sets drawn as `randn(...) * 0.5` (seed 0, samples 1371 and 3975; all ten companion eigenvalues have `|imag| >= 0.078`). Over 4000 samples projected from real scenes it did not occur, but RANSAC draws its minimal samples from mixed inlier/outlier sets, where a sample is random.

## Repro

```python
import torch
from kornia.geometry.epipolar import find_essential

g = torch.Generator().manual_seed(0)
p1 = torch.randn(4000, 5, 2, generator=g, dtype=torch.float64) * 0.5
p2 = torch.randn(4000, 5, 2, generator=g, dtype=torch.float64) * 0.5
Es = find_essential(p1[1371:1372], p2[1371:1372])          # (1, 10, 3, 3)
print("all ten candidates are the identity:", bool((Es == torch.eye(3, dtype=Es.dtype)).all()),
      "| finite:", bool(torch.isfinite(Es).all()), "| Frobenius norm:", Es[0, 0].norm().item())
Eg = find_essential(p1[:1], p2[:1])                        # an ordinary sample for contrast
print("ordinary sample: NaN slots", int(torch.isnan(Eg[0]).flatten(1).any(1).sum()), "of 10")
```

```
kornia kornia 2.14.0
all ten candidates are the identity: True | finite: True | Frobenius norm: 1.7320508075688772
ordinary sample: NaN slots 6 of 10; finite slots have unit norm: True
```

kornia `main` 7ddf731bd, torch 2.14.0, CPU float64 (float32 gives the same), macOS arm64. Same on torch 2.5.1.

## Classification

Silent wrong result. The code comment at the fallback calls it deliberate ("so that nothing below can raise or overflow"), but it makes the no-real-root case the only one whose slots are finite non-solutions, contradicting the per-slot `NaN` contract of the same function. Idiom 4 (degenerate input silently producing output indistinguishable from valid output); kin of the finite-sentinel pattern in #4734.

## Disposition

Focused fix, welcome as a PR: keep the all-NaN candidates (they already flow through `motion_from_essential` and the RANSAC scorer as `NaN`, which every downstream `isfinite` check handles), or document and return an explicit validity mask. Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
